src/services/codefixes/fixAddMissingMember.ts

```diff
@@ -56,43 +56,20 @@ namespace ts.codefix {
             return undefined;
         }
 
-        const classAndMakeStatic = getClassAndMakeStatic(token, checker);
-        if (!classAndMakeStatic) {
-            return undefined;
-        }
-        const { classDeclaration, makeStatic } = classAndMakeStatic;
-        const classDeclarationSourceFile = classDeclaration.getSourceFile();
-        const inJs = isInJavaScriptFile(classDeclarationSourceFile);
-        const call = tryCast(token.parent.parent, isCallExpression);
+        const { parent } = token;
+        if (!isPropertyAccessExpression(parent)) return undefined;
 
-        return { token, classDeclaration, makeStatic, classDeclarationSourceFile, inJs, call };
-    }
+        const leftExpressionType = skipConstraint(checker.getTypeAtLocation(parent.expression));
+        const { symbol } = leftExpressionType;
+        const classDeclaration = symbol && symbol.declarations && find(symbol.declarations, isClassLike);
+        if (!classDeclaration) return undefined;
 
-    function getClassAndMakeStatic(token: Node, checker: TypeChecker): { readonly classDeclaration: ClassLikeDeclaration, readonly makeStatic: boolean } | undefined {
-        const { parent } = token;
-        if (!isPropertyAccessExpression(parent)) {
-            return undefined;
-        }
+        const makeStatic = (leftExpressionType as TypeReference).target !== checker.getDeclaredTypeOfSymbol(symbol);

```

Comment:
@sandersn Is there a better way to detect whether the type was for the static/instance side of the class?